dynamax/linear_gaussian_ssm/inference.py

```diff
@@ -115,14 +143,21 @@ class PosteriorGSSMSmoothed(NamedTuple):
 
 
 # Helper functions
-# _get_params = lambda x, dim, t: x[t] if x.ndim == dim + 1 else x
-def _get_params(x, dim, t):
-    if callable(x):
-        return x(t)
-    elif x.ndim == dim + 1:
-        return x[t]
-    else:
-        return x
+
+    
+def _get_params(params, num_timesteps, t):
+    """Helper function to get parameters at time t."""
+    _get_one_param = lambda t,x: x[t] if x.shape[0]==num_timesteps else x
```

Comment:
Two comments here:

1. While it seems unlikely, I've actually hit bugs from using `num_timesteps` to check if params are time-varying or not. I think I had a test with `T=100` and `D=100` or something... Since it's only the emission covariance matrix that is ambiguous (either 1D or 2D), perhaps we should stick with the dimension-based logic for all the others and then special case the emission cov. If T==D, first issue a warning; then if it's symmetric assume its a DxD covariance matrix, otherwise assume its a TxD array of time-varying diagonal covariance matrices. 

2. This change removes support for callable params functions.

Reply:
- I reverted to the dimension-based logic, although I kept the refactor where `_get_params` fetches all the params at once by called `_get_one_param`. 

- For the ambiguous case where  `ntime==emission_dim`, a warning is issued and the covariance is assumed to be static and non-diagonal.